.storybook/stories-from-readme.js

```diff
@@ -0,0 +1,105 @@
+import * as React from 'react';
+import * as Polaris from '../src';
+import {checkA11y} from '@storybook/addon-a11y';
+import {storiesOf} from '@storybook/react';
+import Playground from '../playground/Playground';
+
+function percyOptions(custom = {}) {
+  const defaults = {skip: true, widths: [375, 1280]};
+  return {...defaults, ...custom};
+}
+
+/**
+ * In most cases we want to test the "All Examples" page as fewer snapshots
+ * means cheaper pricing. However some examples we need to test individually,
+ * usually because they use position:fixed and we don't want examples to
+ * overlay each other as it stops the test being useful.
+ */
+function percyShouldTestIndividualExamples(readmeName) {
+  return ['Modal'].includes(readmeName);
+}
+
+export function generateStories(readme) {
+  // Only generate stories if there are examples
+  if (readme.examples.length === 0) {
+    return;
+  }
+
+  const testIndividualExamples = percyShouldTestIndividualExamples(readme.name);
+
+  storiesOf(`All Components|${readme.name}`, module)
+    .addDecorator(AppProviderDecorator)
+    .addDecorator(checkA11y)
+    .addWithPercyOptions(
+      'All Examples',
+      percyOptions({skip: testIndividualExamples}),
+      () => AllExamplesStoryForReadme(readme),
+    );
+
+  readme.examples.forEach((example) => {
+    storiesOf(`All Components|${readme.name}`, module)
+      .addDecorator(AppProviderDecorator)
+      .addDecorator(checkA11y)
+      .addParameters({
+        // TODO links use styleguide-style URLs. It'd be neat to mutate them
+        // to deeplink to examples in storybook.
+        // TODO remove content-for android/ios blocks from the description
```

Comment:
This could be done using `filterMarkdownForPlatform(markdown, 'web')` ([from the style guide](https://github.com/Shopify/polaris-styleguide/blob/0f7cf4734cbc1df8e4def77c6fe04668a1a86643/server/styleguide-schema/content-parsers/parseExample.ts#L14)).

Reply:
Perfect. I was slightly afraid to delve into styleguide to try and find the right tool. Thanks for pointing me to it.